### PR TITLE
#163519102 CRUD articles 

### PR DESCRIPTION
**Pull request by Nziranziza**
#### What does this PR do?
add CRUD for article
#### Description of Task to be completed?
- create an article
- view an article
- update an article
- delete an article
#### How should this be manually tested?
- git clone
- npm install
- npm run start
- login or create account `base_url/auth`
- create an article `base_url/articles/create`
- update an article `base_url/articles/:articleSlug/edit`

#### What are the relevant pivotal tracker stories?
#163519102

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,5 @@
+CLOUDINARY_API_KEY=
+API_URL=http://localhost:3000/api/v1
+CLOUD_NAME=
+UPLOAD_PRESET=
+UPLOAD_URL=
```
Comment (mcaleb808) [inline .env.example]: Please add a new line here!

**Comment by kimotho-njoki** [comment]
What is going on with the tests?

File: .env.example (hunk 1 above)
Comment (Nziranziza) [reply .env.example]: @mcaleb808 Thank you for your feedback

File: .env.example [hunk 2]
```diff
@@ -1,2 +1,5 @@
-CLOUDINARY_API_KEY=
+CLOUDINARY_API_KEY=sdslkdjslkdjlksdj
```
Comment (mcaleb808) [inline .env.example]: @Nziranziza  our CLOUDINARY_API_KEY is exposed here! please try to remove it!

File: src/__tests__/components/ArticleCreate.test.js [hunk 3]
```diff
@@ -0,0 +1,352 @@
...
+      const videoBtn = wrapper.find('[data-el="video-btn"]');
+      videoBtn.simulate('click');
+      wrapper.find('[data-el="video-input"]')
+        .simulate('change', event);
+      wrapper.find('[data-el="video-input"]');
+      const newState = wrapper.state().editorState;
+      expect(prevState).toEqual(newState);
+      done();
+    });
+    test('should add video', (done) => {
```
Comment (mcaleb808) [inline src/__tests__/components/ArticleCreate.test.js:142]: please add space between tests! not only here

File: .env.example (hunk 2 above)
Comment (Nziranziza) [reply .env.example]: Ok let me handle it

File: src/__tests__/components/ArticleCreate.test.js (hunk 3 above)
Comment (Nziranziza) [reply src/__tests__/components/ArticleCreate.test.js:142]: I am on it thanks!!!

File: src/components/Video/Video.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+      <track kind="captions" />
+    </video>
+  );
+};
+Video.propTypes = {
+  src: PropTypes.string.isRequired,
+  classes: PropTypes.string.isRequired,
+};
+
+export default Video;
```
Comment (kimotho-njoki) [inline src/components/Video/Video.jsx]: @mcaleb808 @Nziranziza @filschristian ????

File: src/redux/actions/loginActions.js [hunk 5]
```diff
@@ -60,11 +60,22 @@ export const validationResponse = (payload, type = INPUT_VALIDATION_FAILURE) =>
   payload,
 });
 
-export const validateCredentials = ({ username, password }) => dispatch => new Promise((resolve) => {
+export const validateCredentials = (
+  { username, password },
+) => dispatch => new Promise((resolve) => {
```
Comment (kimotho-njoki) [inline src/redux/actions/loginActions.js]: `new Promise`? I'm very interested in why we felt we have to explicitly create a promise in this way?

File: src/components/Image/Image.jsx [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const Image = (props) => {
+  const { src, classes } = props;
```
Comment (kimotho-njoki) [inline src/components/Image/Image.jsx:5]: classes? or className? Are you passing many classes or a single className?

File: src/components/Article/ArticlesCurrentUser.jsx [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+  deletingArticle: false,
+  currentUser: {
+    articles: [],
+  },
+};
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps,
+)(ArticlesCurrentUser);
```
Comment (kimotho-njoki) [inline src/components/Article/ArticlesCurrentUser.jsx]: @mcaleb808 @filschristian @Nziranziza? You are all going to pay for this

File: src/__tests__/reducers/articleReducer.test.js [hunk 8]
```diff
@@ -191,4 +208,128 @@ describe('currentUserReducer', () => {
...
+      createArticle: {
+        title: '',
+        body: '',
+        tagList: [expectedState.payload.tag],
+      },
+      articlesList: [],
+    });
+  });
+
+  it('should handle `SUBMIT_ARTICLE_TAG`', () => {
```
Comment (kimotho-njoki) [inline src/__tests__/reducers/articleReducer.test.js:243]: why space out this test but not the rest? Your code needs order

Comment (Nziranziza) [reply src/__tests__/reducers/articleReducer.test.js:243]: Thank you for mentioning @kimotho-njoki. We are going to fix it

File: src/components/Article/ArticlesCurrentUser.jsx (hunk 7 above)
Comment (Nziranziza) [reply src/components/Article/ArticlesCurrentUser.jsx]: We haven't seen this. But thank you for pointing on it we are fixing it

File: src/components/Image/Image.jsx (hunk 6 above)
Comment (Nziranziza) [reply src/components/Image/Image.jsx:5]: Yes, there is the possibility of passing more than one classes. And this is just a variable nothing else. We can still change it if it is confusing but that was our idea.

File: src/components/Video/Video.jsx (hunk 4 above)
Comment (Nziranziza) [reply src/components/Video/Video.jsx]: @kimotho-njoki we could have forgotten it. Let work on it

**Pull request merged**